src/framework/preferences/js/AuxBuilder.js

```diff
@@ -349,10 +341,17 @@ var fluid_2_0 = fluid_2_0 || {};
         return auxSchema;
     };
 
+    // Go over the aux schema to replace terms with their given values.
     fluid.prefs.expandSchema = function (schemaToExpand, indexes, topCommonOptions, elementCommonOptions, mappedDefaults) {
         var auxSchema = fluid.prefs.expandSchemaImpl(schemaToExpand);
         auxSchema.namespace = auxSchema.namespace || "fluid.prefs.created_" + fluid.allocateGuid();
 
+        var terms = fluid.get(auxSchema, "terms");
+        if (terms) {
+            delete auxSchema.terms;
+            fluid.set(auxSchema, ["terms", "terms"], terms);
```

Comment:
The first "terms" is the name of the component that contains a "terms" option to be pass along. The second "terms" is indeed the "terms" option in this component.